### ISSUE TITLE
Restart prompting -- When you just need to start over

### ISSUE THREAD

**Issue by lansingthomas** [atlas-engineer/nyxt#3193]
**Is your feature request related to a problem? Please describe.**
There are cases where a Nyxt restart is required. And we should have an indication to let the user know about it.
- _not to be confused with a page reload_ #3178 

**Describe the solution you'd like**
a prompt buffer to communicate the application-restart-requirement. 

Something like:
---
System changes have been made that require a Nyxt restart:
1. restart Nyxt now
2. restart Nyxt later
3. [Optionally] revert changes

---

**Describe alternatives you've considered**
other apps and browsers use things like badges, or buttons with warning colors. I think that an application-restart-requirement is important_enough to warrant a prompt buffer interruption. And I don't see how we could put a badge or button in the right place easily.

**Additional context**
Thank yoooooouuuuuu!

**Comment by lansingthomas** [comment]
Can I get some help listing the cases where a re-start is required?

- when a theme is changed
- @jmercouris 
- ...

**Comment by lansingthomas** [comment]
anytime the a slot in the class browser is modified
- theme change 
- change renderer
- proxy settings

**Comment by lansingthomas** [comment]
Low priority is fine:

my arguments for this proposal:
1. proposal prevents errors and confusion
2. proposal is consistent with user expectations for apps
3. proposal improves visibility of system status
4. proposal systems should be recognizable _(not rely on users remembering things)_

**Comment by hgluka** [comment]
Yes, I've been thinking about this. I agree that it's a useful feature, but I'm not sure about the implementation. One idea is some kind of indicator on the status bar and the other is a prompt buffer that asks "Do you want to restart now?"

The example I come back to is the Common Settings page:
- As of right now, changing any setting there requires a restart to apply. This means that a "Do you want to restart now?" prompt buffer would pop up any time the user makes a change, which could be a dozen times in one Common Settings "session". That doesn't sound ideal to me.
- And then, even when we finish #3120, we'll give users the option to only apply a setting on restart, so the above problem still stands.

But as @lansingthomas pointed out, there could be other use cases which warrant a prompt buffer interruption.

**Comment by aadcg** [comment]
> My view is that we need to invest our time on the crux of the matter. All parameters from `common-settings` should have immediate effect.

This.
